Scripts/Python/xAvatarCustomization.py

```diff
@@ -2049,11 +2215,48 @@ def __init__(self,clothingType,meshName):
             self.listboxID = kLwrBodyAccLB
         clothinglist = GetAllWithSameGroup(meshName)
         self.clothingItems = []
-        for item in clothinglist:
-            ctype,saturation,inCloset,inClosClr1,inClosClr2 = FindSaturationAndCloset(item[0],item[1])
-            newitem = ClothingItem(item,ctype,saturation,inCloset,inClosClr1,inClosClr2)
+        for newitem in clothinglist:
+            if newitem.isClothingSet:
+                if newitem.clothingSet in clothingSetContents:
+                    hasItem = 0
+                    for tempItem in clothingSetContents[newitem.clothingSet]:
+                        if tempItem.name == newitem.name:
+                            hasItem = 1
+                        
+                    
+                    if not hasItem:
+                        PtDebugPrint('AvaCusta: item ' + newitem.name + ' added to clothing set ' + newitem.clothingSet)
+                        clothingSetContents[newitem.clothingSet].append(newitem)
+                    
+                else:
+                    PtDebugPrint('AvaCusta: adding clothing set ' + newitem.clothingSet + ' and added item ' + newitem.name + ' to it')
+                    clothingSetContents[newitem.clothingSet] = [
+                        newitem]
+            
             # make sure we're not supposed to hide the item
             if CanShowClothingItem(newitem):
+                if newitem.meshicon:
+                    if newitem.groupName in clothingGroupIcons:
+                        PtDebugPrint('AvaCusta: icon ' + newitem.name + ' replaced existing icon ' + clothingGroupIcons[newitem.groupName].name + ' for group ' + newitem.groupName)
+                    else:
+                        PtDebugPrint('AvaCusta: icon ' + newitem.name + ' set for group ' + newitem.groupName)
+                    clothingGroupIcons[newitem.groupName] = newitem
+                elif newitem.groupName in clothingGroupContents:
+                    hasItem = 0
+                    for tempItem in clothingGroupContents[newitem.groupName]:
+                        if tempItem.name == newitem.name:
+                            hasItem = 1
+                            break
+                        
+                    
+                    if not hasItem:
+                        PtDebugPrint('AvaCusta: item ' + newitem.name + ' added to clothing group ' + newitem.groupName)
+                        clothingGroupContents[newitem.groupName].append(newitem)
+                    
```

Comment:
```suggestion
                    for tempItem in clothingGroupContents[newitem.groupName]:
                        if tempItem.name == newitem.name:
                            break
                    else:
                        PtDebugPrint('AvaCusta: item %s added to clothing group %s' + (newitem.name, newitem.groupName))
                        clothingGroupContents[newitem.groupName].append(newitem)
                    
```

`for... else` is an older Python construct that works in very old python versions. The `else:` clause will only be executed if no `break` is encountered during the for loop.